src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -130,12 +130,33 @@ function getSpaceAvailable(storagePools, connectionName) {
     return space;
 }
 
+function getVmName(vmName, connectionName, vms, os) {
+    if (!isEmpty(vmName.trim()))
+        return vmName;
+
+    let retName = connectionName;
```

Comment:
It's redundant to include the connection name into the domain name. These are already namespaced, and it will make them even longer.